extensions/wikia/PortableInfobox/services/PortableInfoboxDataService.class.php

```diff
@@ -63,4 +58,68 @@ public function getImages() {
 
 		return array_keys( $images );
 	}
+
+	/**
+	 * @desc For given Title, get property 'infoboxes' from parser output. If property is empty, this may mean that
+	 * template is inside the <noinclude> tag. In this case, we want to skip the <includeonly> tags, get from this only
+	 * infoboxes and parse them again to check their presence and get params.
+	 * @param $title Title
+	 * @return mixed
+	 */
+	protected function getParsedInfoboxes( $title ) {
+		$article = Article::newFromTitle( $title, RequestContext::getMain() );
+
+		if ( $title->getNamespace() === NS_TEMPLATE ) {
+			$templateText = $article->fetchContent();
+			$includeonlyText = $this->getIncludeonlyText( $templateText );
+
+			if ( $includeonlyText ) {
+				$parser = new Parser();
+				$parserOptions = new ParserOptions();
+				$frame = $parser->getPreprocessor()->newFrame();
+
+				$templateTextWithoutIncludeonly = $parser->getPreloadText( $includeonlyText, $article->getTitle(), $parserOptions );
+				$infoboxes = $this->getInfoboxes( $templateTextWithoutIncludeonly );
+
+				if ( $infoboxes ) {
+					foreach ( $infoboxes as $infobox ) {
+						PortableInfoboxParserTagController::getInstance()->render( $infobox, $parser, $frame );
+					}
+					return $parser->getOutput()->getProperty( self::INFOBOXES_PROPERTY_NAME );
+				}
+			}
+		}
+
+		//on empty parser cache this should be regenerated, see WikiPage.php:2996
+		$parserOutput = $article->getParserOutput();
+		$parsedInfoboxes = $parserOutput ?
+			$parserOutput->getProperty( self::INFOBOXES_PROPERTY_NAME )
+			: false;
+
+		return $parsedInfoboxes;
```

Comment:
we need to cover the case when article is a template, contains `includeonly` tags and inside them we have one or more infoboxes we want to get the params from. The includeonly inside templates are not getting parsed in the Template page (they are parsed only when parsing invocations of them inside articles) thus, we need to manually run parser for the separated infoboxes found in the `includeonly` tags.